compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/MethodGenUtils.java

```diff
@@ -123,6 +123,11 @@ static String getFrameClassName(String pkgName, String funcName, BType attachedT
     }
 
     private static boolean isValidType(BType attachedType) {
-        return attachedType != null && (attachedType.tag == TypeTags.OBJECT || attachedType.tag == TypeTags.RECORD);
+        if (attachedType != null) {
+            BType referredAttachedType = JvmCodeGenUtil.getReferredType(attachedType);
```

Comment:
Lets update `JvmCodeGenUtil.getReferredType()` to return `null` if the type is null